src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -589,6 +635,57 @@ void FindEnzymesDialog::accept() {
     QDialog::accept();
 }
 
+void FindEnzymesDialog::sl_setEnzymesBySuppliers(QStringList checkedSuppliers) {
+    const auto enzymes = EnzymesSelectorWidget::getLoadedEnzymes();
+    QList<SEnzymeData> visibleEnzymes;
+    for (const auto& enzyme : qAsConst(enzymes)) {
+        for (const auto& supplier : qAsConst(enzyme->suppliers)) {
+            CHECK_CONTINUE(checkedSuppliers.contains(supplier));
+
+            visibleEnzymes.append(enzyme);
+            break;
+
+        }
+    }
+    enzSel->setEnzymesList(visibleEnzymes);
+    AppContext::getSettings()->setValue(EnzymeSettings::CHECKED_SUPPLIERS, checkedSuppliers.join(SUPPLIERS_LIST_SEPARATOR));
+}
+
+void FindEnzymesDialog::sl_updateSuppliers() {
+    const auto& loadedSuppliers = EnzymesSelectorWidget::getLoadedSuppliers();
+    cbSuppliers->clear();
+    for (const auto& supplier : qAsConst(loadedSuppliers)) {
+        cbSuppliers->addItem(supplier);
```

Comment:
Use `addItems` to add all suppliers in 1 line and have only 1 set of change signals as the result.

Reply:
OK